src/NuGet.Core/NuGet.Packaging/TopologicalSortUtility.cs

```diff
@@ -38,19 +38,25 @@ public static class TopologicalSortUtility
 
         private static int GetParentCount(List<PackageDependencyInfo> packages, string id)
         {
-            int count = 0;
+            var parentCount = 0;
+            var count = packages.Count;
 
-            foreach (var package in packages)
+            for (var i = 0; i < count; i++)
```

Comment:
It won't make a difference in the scheme of things here. The `foreach` is cleaner and less prone to bugs; I'd prefer to use it unless it's observed to have non-negligible overhead in practice (keep in mind I'm testing this on _very_ large projects, so if it doesn't appear as a problem in my testing then likely no one will be able to find it).